File: test-suite-function-client-aws/src/test/java/io/micronaut/docs/function/client/aws/atnamed/AnalyticsClient.java
package io.micronaut.docs.function.client.aws.atnamed;

import io.micronaut.function.client.FunctionClient;
import jakarta.inject.Named;
@FunctionClient
public interface AnalyticsClient {

    @Named("analytics") // <1>
    String visit(String productId);
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/function/client/aws/atnamed/AnalyticsClient.kt
package io.micronaut.docs.function.client.aws.atnamed

//tag::clazz[]
import io.micronaut.function.client.FunctionClient
import jakarta.inject.Named

@FunctionClient
internal interface AnalyticsClient {
    @Named("analytics") // <1>
    fun visit(productId: String): String
}
//end::clazz[]